### PR TITLE
fix most high and medium coverity issues

### PR DESCRIPTION
**Pull request by rileyberton**

### CODE REVIEW

File: src/eventer/eventer_jobq.c [hunk 1]
```diff
@@ -209,9 +204,15 @@ eventer_jobq_create_internal(const char *queue_name, eventer_jobq_memory_safety_
     stats_rob_i32(jobq_ns, "backlog", (void *)&jobq->backlog);
     stats_rob_i64(jobq_ns, "timeouts", (void *)&jobq->timeouts);
   }
- out:
   pthread_mutex_unlock(&all_queues_lock);
   return jobq;
+
+ error_out:
+  free((void *)jobq->queue_name);
```
Comment (rileyberton) [inline src/eventer/eventer_jobq.c]: Leaked jobq on failure:

https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140600&defectInstanceId=6723254&mergedDefectId=1452102

File: src/luamtev.c [hunk 2]
```diff
@@ -282,5 +282,15 @@ int main(int argc, char **argv, char **envp) {
   mtev_main(APPNAME, config_file, debug, foreground,
             MTEV_LOCK_OP_LOCK, NULL, droptouser, droptogroup,
             child_main);
+
+  free(function);
```
Comment (rileyberton) [inline src/luamtev.c]: Bunch of leaks:

Example: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140689&defectInstanceId=6723260&mergedDefectId=1452114

File: src/modules/lua.c [hunk 3]
```diff
@@ -827,7 +827,7 @@ static char *
 package_manip_path(char *in, const char *find, const char *replace) {
   char *ocp, *npath = in;
   if(NULL != (ocp = strstr(in, find))) {
-    int nlen = strlen(in) + strlen(replace);
+    int nlen = strlen(in) + strlen(replace) + 1; //NUL term
```
Comment (rileyberton) [inline src/modules/lua.c]: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140696&defectInstanceId=6723248&mergedDefectId=1376693

File: src/modules/lua_general.c [hunk 4]
```diff
@@ -143,17 +143,22 @@ lua_general_new_resume_info(lua_module_closure_t *lmc) {
...
 lua_general_handler_ex(mtev_dso_generic_t *self,
                       const char *module, const char *function) {
+  char errbuf[128];
   int status, rv;
-  lua_general_conf_t *conf = get_config(self);
-  lua_module_closure_t *lmc = pthread_getspecific(conf->key);
-  mtev_lua_resume_info_t *ri = NULL;
   const char *err = NULL;
-  char errbuf[128];
+  mtev_lua_resume_info_t *ri = NULL;
```
Comment (rileyberton) [inline src/modules/lua_general.c]: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140701&defectInstanceId=6723271&mergedDefectId=1452095

File: src/mtev_conf.c [hunk 5]
```diff
@@ -1315,6 +1315,9 @@ mtev_conf_section_to_xpath(mtev_conf_section_t* section) {
     } else {
       buff_len = current_entry_len + sizeof("/descendant::[999999]\0");
       current_entry = malloc(buff_len);
+      if (buff_len != 512) {
+        free(buff);
+      }
       buff = current_entry;
```
Comment (postwait) [inline src/mtev_conf.c]: I think this is ugly and fragile.  Could we instead have a `char *onheap = NULL;` that is assigned in here as `onheap = buff = current_entry;` and free(onheap)?

File: src/modules/lua_mtev.c [hunk 6]
```diff
@@ -4044,6 +4044,7 @@ mtev_lua_serialize(lua_State *L, int index){
     case(LUA_TNIL): // we already returned NULL
     default: 
       free(data);
+      data = NULL;
```
Comment (rileyberton) [inline src/modules/lua_mtev.c]: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140716&defectInstanceId=6723289&mergedDefectId=1376743

File: src/modules/lua_web.c [hunk 7]
```diff
@@ -158,18 +158,23 @@ static int
...
-  lua_web_conf_t *conf = the_one_conf;
-  lua_module_closure_t *lmc = mtev_lua_web_setup_lmc(conf->self);
   mtev_lua_resume_info_t *ri;
   mtev_lua_resume_rest_info_t *ctx = NULL;
   lua_State *L;
   eventer_t conne = NULL;
-  mtev_http_response *res = mtev_http_session_response(restc->http_ctx);
 
-  if(!lmc || !conf) {
+  lua_web_conf_t *conf = the_one_conf;
```
Comment (rileyberton) [inline src/modules/lua_web.c]: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140699&defectInstanceId=6723275&mergedDefectId=1452110

File: src/mtev_cluster.c [hunk 8]
```diff
@@ -714,7 +714,9 @@ int mtev_cluster_update_internal(mtev_conf_section_t cluster,
   new_cluster->period = period;
   new_cluster->timeout = timeout;
   new_cluster->maturity = maturity;
-  qsort(nlist, n_nodes, sizeof(*nlist), mtev_cluster_node_compare);
+  if (nlist != NULL) {
```
Comment (rileyberton) [inline src/mtev_cluster.c]: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140461&defectInstanceId=6723226&mergedDefectId=1376675

File: src/mtev_cluster.c [hunk 9]
```diff
@@ -912,6 +914,7 @@ mtev_cluster_set_heartbeat_payload(mtev_cluster_t *cluster,
   }
 
   if(payload_len_sum + payload_length > MAX_PAYLOAD_LEN_SUM) {
+    free(hash_key);
```
Comment (rileyberton) [inline src/mtev_cluster.c]: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140461&defectInstanceId=6723269&mergedDefectId=1376719

File: src/mtev_cluster.c [hunk 10]
```diff
@@ -1059,7 +1062,7 @@ rest_show_cluster_json(mtev_http_rest_closure_t *restc, int n, char **p) {
   goto cleanup;
 
  cleanup:
-  if(doc) MJ_DROP(doc);
+  MJ_DROP(doc);
```
Comment (rileyberton) [inline src/mtev_cluster.c]: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140461&defectInstanceId=6723324&mergedDefectId=1452108

File: src/mtev_conf.c [hunk 11]
```diff
@@ -1148,7 +1148,7 @@ mtev_conf_magic_mix(const char *parentfile, xmlDocPtr doc, include_node_t* inc_n
       if((include_nodes[inc_idx].doc) || (include_nodes[inc_idx].snippet)) {
         xmlNodePtr n, more_kids;
         mtev_conf_magic_mix(infile, include_nodes[inc_idx].doc, &(include_nodes[inc_idx]));
-        strncpy(include_nodes[inc_idx].path, infile, sizeof(include_nodes[inc_idx].path));
+        strncpy(include_nodes[inc_idx].path, infile, sizeof(include_nodes[inc_idx].path) - 1); // room for NUL
```
Comment (rileyberton) [inline src/mtev_conf.c]: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140502&defectInstanceId=6723197&mergedDefectId=1376651

File: src/mtev_conf.c [hunk 12]
```diff
@@ -1315,6 +1315,9 @@ mtev_conf_section_to_xpath(mtev_conf_section_t* section) {
     } else {
       buff_len = current_entry_len + sizeof("/descendant::[999999]\0");
       current_entry = malloc(buff_len);
+      if (buff_len != 512) {
```
Comment (rileyberton) [inline src/mtev_conf.c]: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140502&defectInstanceId=6723322&mergedDefectId=1376701

File: src/mtev_conf.c [hunk 13]
```diff
@@ -1786,8 +1789,11 @@ mtev_conf_property_iter(mtev_conf_section_t section,
 
 int
 mtev_conf_set_string(mtev_conf_section_t section,
-                     const char *path, const char *value) {
+                     const char *path, const char *value) 
+{
+  mtev_conf_section_t *sections = NULL;
   xmlNodePtr current_node = (xmlNodePtr)section;
```
Comment (rileyberton) [inline src/mtev_conf.c]: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140502&defectInstanceId=6723303&mergedDefectId=1376702

File: src/mtev_reverse_socket.c [hunk 14]
```diff
@@ -294,6 +294,28 @@ static void APPEND_OUT(reverse_socket_t *rc, reverse_frame_t *frame_to_copy) {
...
+  id = frame->channel_id & 0x7fff;
+  rc->data.channels[id].out_bytes += frame->buff_len;
+  rc->data.channels[id].out_frames += 1;
+  if(rc->data.outgoing_tail) {
+    rc->data.outgoing_tail->next = frame;
+    rc->data.outgoing_tail = frame;
+  }
+  else {
+    rc->data.outgoing = rc->data.outgoing_tail = frame;
+  }
```
Comment (rileyberton) [inline src/mtev_reverse_socket.c]: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140525&defectInstanceId=6723295&mergedDefectId=1452109

File: src/mtev_reverse_socket.c [hunk 15]
```diff
@@ -502,6 +525,10 @@ mtev_reverse_socket_handler(eventer_t e, int mask, void *closure,
 socket_error:
     /* Exceptions cause us to simply snip the connection */
     mtevL(nldeb, "%s reverse_socket: %s\n", rc->id, socket_error_string);
+    if (needs_unlock) {
```
Comment (rileyberton) [inline src/mtev_reverse_socket.c]: https://scan7.coverity.com/reports.htm#v20002/p12027/fileInstanceId=30140525&defectInstanceId=6723297&mergedDefectId=1376685

File: src/eventer/eventer_jobq.c [hunk 16]
```diff
@@ -209,9 +204,15 @@ eventer_jobq_create_internal(const char *queue_name, eventer_jobq_memory_safety_
...
+  free((void *)jobq);
+  pthread_mutex_unlock(&all_queues_lock);
```
Comment (aidancully) [inline src/eventer/eventer_jobq.c]: do we need to `pthread_mutex_destory`, `sem_destroy`, etc., if the error occurred at those points?

Comment (rileyberton) [inline src/eventer/eventer_jobq.c]: dunno.. it's a potential leak that never really happens.  So the chances are low we ever even hit this label

Comment (aidancully) [inline src/eventer/eventer_jobq.c]: patch is certainly an improvement, so doesn't block the merge.

**Pull request merged**